.github/workflows/main.yml

```
@@ -31,11 +31,12 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Set up JDK 21
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
```

actionlint (1.7.10)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)